va/va.go

```diff
@@ -486,80 +523,117 @@ func (va *ValidationAuthorityImpl) performRemoteValidation(
+			currProb, err = bgrpc.PBToProblemDetails(resp.result.Problems)
 			if err != nil {
-				va.log.Errf("Remote VA %q.PerformValidation returned malformed problem: %s", res.hostname, err)
+				va.log.Errf("Remote VA %q.PerformValidation returned malformed problem: %s", perspective, err)
 				currProb = probs.ServerInternal("Remote PerformValidation RPC returned malformed result")
 			}
 		} else {
-			good++
+			// The remote VA returned a successful result.
+			passed = append(passed, resp.rva)
 		}
 
 		if firstProb == nil && currProb != nil {
 			firstProb = currProb
 		}
 
-		// Return as soon as we have enough successes or failures for a definitive result.
-		if good >= required {
-			return nil
+		if len(passed) >= required {
+			// Enough successful responses to reach quorum.
+			return va.summarizeMPIC(passed, failed), nil
 		}
-		if bad > va.maxRemoteFailures {
+		if len(failed) > va.maxRemoteFailures {
+			// Too many failed responses to reach quorum.
 			va.metrics.remoteValidationFailures.Inc()
 			firstProb.Detail = fmt.Sprintf("During secondary validation: %s", firstProb.Detail)
-			return firstProb
+			return va.summarizeMPIC(passed, failed), firstProb
 		}
 
 		// If we somehow haven't returned early, we need to break the loop once all
 		// of the VAs have returned a result.
-		if good+bad >= len(va.remoteVAs) {
+		if len(passed)+len(failed) >= remoteVACount {
 			break
 		}
 	}
 
-	// This condition should not occur - it indicates the good/bad counts neither
-	// met the required threshold nor the maxRemoteFailures threshold.
-	return probs.ServerInternal("Too few remote PerformValidation RPC results")
+	// This condition should not occur - it indicates the passed/failed counts
+	// neither met the required threshold nor the maxRemoteFailures threshold.
+	return va.summarizeMPIC(passed, failed), probs.ServerInternal("Too few remote PerformValidation RPC results")
+}
+
+func (va *ValidationAuthorityImpl) summarizeMPIC(passed, failed []RemoteVA) *mpicSummary {
+	var passedPerspectives []string
+	var failedPerspectives []string
+	var passedRIRs []string
+	for _, rva := range passed {
+		passedPerspectives = append(passedPerspectives, rva.Perspective)
+		passedRIRs = append(passedRIRs, rva.RIR)
+	}
+	for _, rva := range failed {
+		failedPerspectives = append(failedPerspectives, rva.Perspective)
+		passedRIRs = append(passedRIRs, rva.RIR)
+	}
+
+	if passedPerspectives == nil {
+		passedPerspectives = []string{}
+	}
+	if failedPerspectives == nil {
+		failedPerspectives = []string{}
+	}
+	if passedRIRs == nil {
+		passedRIRs = []string{}
+	}
+
+	return &mpicSummary{
+		PassedPerspectives: passedPerspectives,
+		FailedPerspectives: failedPerspectives,
+		PassedRIRs:         passedRIRs,
+		// Note: len(va.remoteVAs) is greater than len(passed) + len(failed) because some
+		// are canceled on reaching quorum.
+		QuorumResult: fmt.Sprintf("%d/%d", len(passed), len(va.remoteVAs)),
```

Comment:
If we make an "attempt" I believe that we have to log the outcome of that attempt according to the BRs:


### 5.4.1 Types of events recorded

...

The CA SHALL record at least the following events:

...

Multi-Perspective Issuance Corroboration attempts from each Network Perspective, minimally recording the following information:
- a. an identifier that uniquely identifies the Network Perspective used;
- b. the attempted domain name and/or IP address; and
- c. the result of the attempt (e.g., "domain validation pass/fail", "CAA permission/prohibition").>
Multi-Perspective Issuance Corroboration quorum results for each attempted domain name or IP address represented in a Certificate request (i.e., "3/4" which should be interpreted as "Three (3) out of four (4) attempted Network Perspectives corroborated the determinations made by the Primary Network Perspective).